relay-server/src/envelope.rs

```diff
@@ -1174,6 +1174,14 @@ impl Envelope {
         }
     }
 
+    /// Returns `true` whether the event is fully normalized, `false`.
+    pub fn event_fully_normalized(&self) -> bool {
```

Comment:
I would not put normalization concerns and business logic on the envelope.

If we have a newtype, we also have a place to put it.

Reply:
The `fully_normalized` flag is already on the envelope, so IMO it's fine to have this getter.